hpy/devel/src/runtime/buildvalue.c

```diff
@@ -97,23 +97,24 @@ HPyAPI_HELPER
 HPy HPy_BuildValue(HPyContext *ctx, const char *fmt, ...)
 {
     va_list values;
+    HPy result;
     va_start(values, fmt);
     HPy_ssize_t size = count_items(ctx, fmt, '\0');
     if (size < 0) {
-        return HPy_NULL;
+        result = HPy_NULL;
     } else if (size == 0) {
-        return HPy_Dup(ctx, ctx->h_None);
+        result = HPy_Dup(ctx, ctx->h_None);
     } else if (size == 1) {
         int needs_close;
-        HPy result = build_single(ctx, &fmt, &values, &needs_close);
+        result = build_single(ctx, &fmt, &values, &needs_close);
         if (!needs_close) {
-            return HPy_Dup(ctx, result);
-        } else {
-            return result;
+            result = HPy_Dup(ctx, result);
         }
     } else {
-        return build_tuple(ctx, &fmt, &values, size, '\0');
+        result = build_tuple(ctx, &fmt, &values, size, '\0');
     }
+    va_end(values);
```

Comment:
This is the same problem as #73. We should run cppcheck in any version since it prevents these kind of errors which could escalate to security issues.

Reply:
Not sure if running cppcheck in _any_ version makes sense but we should at least run a decent version of it. It can also make sense to run several carefully picked versions.
However, the main problem currently is that `ubuntu-latest` in the CI provides a rather old version. We will need to manually download and build it. But that again imposes a risk since since external changes could then break our checks (any in my experience, this _will_ happen).
Anyway, please feel free to open PRs to improve the situation. I'd really appreciate that.